pkg/controller/rollout/rollout_releaseManager.go

```diff
@@ -10,4 +10,5 @@ type ReleaseManager interface {
 	doCanaryFinalising(c *RolloutContext) (bool, error)
 	fetchBatchRelease(ns, name string) (*v1beta1.BatchRelease, error)
 	removeBatchRelease(c *RolloutContext) (bool, error)
+	doCanaryReset(c *RolloutContext) (bool, error)
```

Comment:
what is the difference bewteen doCanaryFinalising  and doCanaryReset,  plz add the missing description of all these methods, and the workflow to call these methods

Reply:
comment added